src/lib/php/Report/ClearedGetterCommon.php

```diff
@@ -163,26 +170,35 @@ protected function groupStatements($ungrupedStatements, $extended, $agentcall, $
           $statements[] = $singleStatement;
         }
       }
-      if (!empty($statement['textfinding']) && !empty($agentcall)) {
-        $findings[$fileName] = array(
+      if (!empty($statement['textfinding']) && !empty($agentCall) && $agentCall != "license") {
+        $findings[] = array(
+            "licenseId" => $licenseId,
             "content" => convertToUTF8($statement['textfinding'], false),
             "text" => convertToUTF8($text, false),
             "files" => array($fileName)
           );
         if ($extended) {
-          $findings[$fileName]["comments"] = convertToUTF8($comments, false);
+          $key = array_search($statement['textfinding'], array_column($findings, 'content'));
+          $findings[$key]["comments"] = convertToUTF8($comments, false);
         }
       }
-    }
-    if ($agentcall == "copyright" && $isUnifiedReport == true) {
-      if (!empty($findings)) {
-        return array("userFindings" => $findings, "scannerFindings" => $statements);
-      } else {
-        return array("scannerFindings" => $statements);
+      //To keep the schedular alive for large files
+      $countLoop += 1;
+      if (is_int($countLoop/500)) {
```

Comment:
A simpler, more common method will be

```suggestion
      if ($countLoop % 500 == 0) {
```